internal/archive/archive.go

```diff
@@ -81,7 +81,12 @@ func GenerateTarWithWriter(genFn func(TarWriter) error, twf TarWriterFactory) io
 		errChan <- closeErr
 	}()
 
+	closed := false
 	return ioutils.NewReadCloserWrapper(pr, func() error {
+		if closed {
```

Comment:
I forget the exact details, but there was problem that occurs when this gets closed too many times, but there was no formal error surfacing, which made it hard to debug.

Reply:
I think we'd want to investigate this a tad further. At least have a bit better understanding as to why this was necessary. I'm afraid this could have some adverse effects. If I'm not mistaken we use this strategy in various places so making this change only here either means it's "broken" in multiple places or this particular implementation has issues.

Reply:
I'll try reverting it and see if it's still needed. It was part of the original implementation, and the PR has evolved quite a bit since then (though I don't believe it has changed much in the area we needed this).

Reply:
Deferred to https://github.com/buildpacks/pack/issues/792